### DISCUSSION TITLE
how to choose different speaker

### DISCUSSION THREAD

**Discussion by wanghaisheng** [coqui-ai/TTS#733] [General Q&A]
in the demo page there are dozens of them 

https://erogol.github.io/ddc-samples/
how to choose them speficically in the https://colab.research.google.com/drive/1tKHSI20kRlOL0PSA8mCVJQIrgRIswg0F?usp=sharing#scrollTo=qqDhEPEH10uh

```
align, spec, stop_tokens, wav = tts(model, sentence, TTS_CONFIG, use_cuda, ap, use_gl=False, figures=True)
```

**Comment by Sadam1195** [comment]
you can list models using `tts --list_models` and choose one you like and run inference like this 
```
tts --text "Text for TTS" \
    --model_name "<type>/<language>/<dataset>/<model_name>" \
    --vocoder_name "<type>/<language>/<dataset>/<model_name>" \
    --out_path folder/to/save/output.wav
```
@wanghaisheng

**Comment by wanghaisheng** [comment]
actually  unless I download them all and run test one by one, I will not know the mapping between the demo page sound between model name.
I am now awared each different speaker is one model separately.

**Reply by mariusa** [reply]
+1

It would be easier if the "TTS Models" link at
https://coqui.ai/models
will also have 10-20 sec samples for each model. Recorded, ready to play, not links to  https://colab.research.google.com which new potential users evaluating Coqui don't know how to run (I'm stuck there).

**Reply by erogol** [reply]
It doesn't make sense since after listening to the model you still need to run the model which requires you to run `tts` anyhow. 

You don't need to get stuck too. Here is the help https://tts.readthedocs.io/en/latest/tutorial_for_nervous_beginners.html

**Reply by wanghaisheng** [reply]
let me clarify. each model is about  above 300M .one by one is boring.
what we want to know is  after listen the demo sound over website, got to know the exact model we need instead of download them all and do experiments by myself

**Reply by wanghaisheng** [reply]
your link doc really dont help I have manually download model following colab code instructions  as following
```
dubo@LAPTOP-I2R9VG58 MINGW64 /d/Download/audio-visual/auto_video-jigsaw/tts/coqui-tts
$ ls
multispeaker_vctk_tacotron2_ddc_wavegrad.py  tts_config.json    tts_scale_stats.npy  vocoder_model.pth.tar
speakers.json                                tts_model.pth.tar  vocoder_config.json  vocoder_scale_stats.npy
(torch) 
dubo@LAPTOP-I2R9VG58 MINGW64 /d/Download/audio-visual/auto_video-jigsaw/tts/coqui-tts
$ tts --text "Text for TTS"     --model_name "tts_model.pth.tar"     --vocoder_name "vocoder_model.pth.tar"     --out_path output.wav
Traceback (most recent call last):
  File "d:\program files\anaconda3\envs\torch\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "d:\program files\anaconda3\envs\torch\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "D:\Program Files\anaconda3\envs\torch\Scripts\tts.exe\__main__.py", line 7, in <module>
  File "d:\program files\anaconda3\envs\torch\lib\site-packages\TTS\bin\synthesize.py", line 205, in main
    model_path, config_path, model_item = manager.download_model(args.model_name)
  File "d:\program files\anaconda3\envs\torch\lib\site-packages\TTS\utils\manage.py", line 99, in download_model   
    model_type, lang, dataset, model = model_name.split("/")
ValueError: not enough values to unpack (expected 4, got 1)
```
@erogol

**Comment by wanghaisheng** [comment]
@mariusa  thanks for support me
this is really wonderful project
these little doc work can help a lot  guy not that code expert like me
